text/000-deprecation-native-function-prototype-extensions.md

```diff
@@ -0,0 +1,65 @@
+- Start Date: 2017-11-20
+- RFC PR: https://github.com/emberjs/rfcs/pull/272
+- Ember Issue:
+
+# Deprecate Function.prototype.on, Function.prototype.observes and Function.prototype.property
+
+## Summary
+
+This RFC proposes to deprecate `Function.prototype.on`,
+`Function.prototype.observes` and `Function.prototype.property`
+
+## Motivation
+
+Ember has been moving away from extending native prototypes due to the confusion
+that this causes users: is it specifically part of Ember, or JavaScript?
+
+Continuing in that direction, we should consider recommending the usage of
+[`on` (`@ember/object/evented`)](https://emberjs.com/api/ember/2.18/classes/@ember%2Fobject%2Fevented/methods/on?anchor=on), [`observer` (`@ember/object`)](https://emberjs.com/api/ember/2.18/classes/@ember%2Fobject/methods/observer?anchor=observer) and [`computed` (`@ember/object`)](https://emberjs.com/api/ember/2.18/classes/@ember%2Fobject/methods/computed?anchor=computed) as opposed to their native
+prototype extension equivalents.
+We go from two ways to do something, to one.
+
+[`eslint-plugin-ember` already provides this as a rule](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-function-prototype-extensions.md).
+
+## Transition Path
+
+The replacement functionality already exists in the form of `on`, `observer`, and `computed`.
+We don't need to build anything new specifically, however, the bulk of the transition will be
+focused on deprecating the native prototype extensions.
+
+Borrowing from the [ESLint plugin example](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-function-prototype-extensions.md):
```

Comment:
I'd like to suggest something here. Keep "Transtition Path" focused on the implementation details of the deprecations, and move the "before"/"after" code samples to "How We Teach This".